pkg/machines/components/desktopConsole.jsx

```diff
@@ -0,0 +1,219 @@
+        '<i>Remote Viewer</i>');
+
+    const msg2 = cockpit.format(_("$0 is available for most operating systems. To install it, search for it in GNOME Software or run the following:"),
+        '<i>Remote Viewer</i>');
+
+    const downloadMsg = cockpit.format(_("Download the MSI from $0"),
+        '<a href="https://virt-manager.org/download/" target="_blank">virt-manager.org</a>');
+
+    return (
+        <div>
+            <br />
+            <p className='machines-desktop-more-info-text' dangerouslySetInnerHTML={{__html: msg1}} />
+            <p className='machines-desktop-more-info-text' dangerouslySetInnerHTML={{__html: msg2}} />
+
+            <ul className='machines-desktop-install-instructs'>
+                <MoreInformationInstallVariant os='RHEL, CentOS' command='sudo yum install virt-viewer'/>
+                <MoreInformationInstallVariant os='Fedora' command='sudo dnf install virt-viewer'/>
+                <MoreInformationInstallVariant os='Ubuntu, Debian' command='sudo apt-get install virt-viewer'/>
+                <MoreInformationInstallVariant os='Windows' innerHtml={downloadMsg}/>
+            </ul>
+        </div>
+    );
+}
+
+class MoreInformation extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            expanded: false,
+        };
+
+        this.onClick = this.onClick.bind(this);
+        this.getContent = this.getContent.bind(this);
+    }
+
+    onClick() {
+        this.setState({
+            expanded: !this.state.expanded,
+        });
+    }
+
+    getContent() {
+        const { vm } = this.props;
+        const { provider, providerState } = this.props.config;
+
+        let content = <MoreInformationContent />;
+        if (provider.consoleClientResourcesFactory) {
+            // external provider can have specific instructions for console setup
+            const ProviderConsoleClientResources = provider.consoleClientResourcesFactory(vm, providerState);
+            if (ProviderConsoleClientResources) {
+                content = <ProviderConsoleClientResources displays={vm.displays} />;
+            }
+        }
+
+        return content;
+    }
+
+    render() {
+        if (!this.state.expanded) {
+            return (
+                <a href='#' onClick={this.onClick}>
+                    <span className='fa fa-angle-right'/>&nbsp;
+                    {_("More Information")}
+                </a>);
+        }
+
+        return (
+            <div>
+                <a href='#' onClick={this.onClick}>
+                    <span className='fa fa-angle-down'/>&nbsp;
+                    {_("More Information")}
+                </a>
+                {this.getContent()}
+            </div>);
+    }
+}
+
+const ConnectWithRemoteViewer = ({ vm, config, onDesktopConsole }) => {
+    let display = vm.displays.spice;
+    if (!display) {
+        display = vm.displays.vnc;
+    }
+    const onLaunch = () => onDesktopConsole(display)
```

Comment:
Good catch, accidentally missed. It's another ES6 syntax. I will add the semicolon.